packages/server/src/fhircast/routes.ts

```diff
@@ -71,65 +71,42 @@ publicSTU3Routes.get('/.well-known/fhircast-configuration', (_req: Request, res:
 });
 
 // Register a new subscription
+// Or publish an event depending on payload
 protectedCommonRoutes.post(
   '/',
-  [
-    body('hub.channel.type').notEmpty().withMessage('Missing hub.channel.type'),
-    body('hub.mode').notEmpty().withMessage('Missing hub.mode'),
-    body('hub.topic').notEmpty().withMessage('Missing hub.topic'),
-    body('hub.events').notEmpty().withMessage('Missing hub.events'),
-  ],
+  oneOf(
+    [
+      [
+        body('id').notEmpty().withMessage('Missing event ID'),
+        body('timestamp').notEmpty().withMessage('Missing event timestamp'),
+        body('event').notEmpty().withMessage('Missing event payload'),
+        body('event["hub.topic"]').notEmpty().withMessage('Missing event["hub.topic"]'),
+        body('event["hub.event"]').notEmpty().withMessage('Missing event["hub.event"]'),
+        body('event.context').notEmpty().withMessage('Missing event.context'),
+      ],
+      [
+        body('hub.channel.type').notEmpty().withMessage('Missing hub.channel.type'),
+        body('hub.mode').notEmpty().withMessage('Missing hub.mode'),
+        body('hub.topic').notEmpty().withMessage('Missing hub.topic'),
+        body('hub.events').notEmpty().withMessage('Missing hub.events'),
+      ],
+    ],
+    { errorType: 'least_errored' }
+  ),
   asyncWrap(async (req: Request, res: Response) => {
     const errors = validationResult(req);
     if (!errors.isEmpty()) {
       sendOutcome(res, invalidRequest(errors));
       return;
     }
 
-    const type = req.body['hub.channel.type'];
-    if (type !== 'websocket') {
-      sendOutcome(res, badRequest('Invalid hub.channel.type'));
-      return;
-    }
-
-    const mode = req.body['hub.mode'] as 'subscribe' | 'unsubscribe';
-    if (!(['subscribe', 'unsubscribe'] as const).includes(mode)) {
-      sendOutcome(res, badRequest('Invalid hub.mode'));
+    // If there is an ID, this is a context change request
+    if (req.body.id) {
```

Comment:
We check for `id` to determine whether its a context change request or subscription request